instrumentation/jmx-metrics/javaagent/src/main/java/io/opentelemetry/instrumentation/javaagent/jmx/JmxMetricInsightInstaller.java

```diff
@@ -81,10 +80,15 @@ private static void buildFromDefaultRules(
   private static void buildFromUserRules(
       MetricConfiguration conf, ConfigProperties configProperties) {
     List<String> configFiles = configProperties.getList("otel.jmx.config");
+    String configFileUrlString;
     for (String configFile : configFiles) {
-      JmxMetricInsight.getLogger().log(FINE, "JMX config file name: {0}", configFile);
+      configFileUrlString = configFile;
+      if (configFile != null && !configFile.contains("://")) {
+        configFileUrlString = "file://" + configFile;
+      }
+      JmxMetricInsight.getLogger().log(FINE, "JMX config file name: {0}", configFileUrlString);
       RuleParser parserInstance = RuleParser.get();
-      try (InputStream inputStream = Files.newInputStream(Paths.get(configFile))) {
+      try (InputStream inputStream = new URL(configFileUrlString).openStream()) {
```

Comment:
I think it would be also useful to have some kind of retry mechanism if reading remote config fails due to temporary network issues. Unfortunately this must be implemented in a way that would not affect agent startup time, that @SylvainJuge already pointed out.

Reply:
@robsunday it's not possible to implement a retry mechanism without affecting agent startup time without a broader change..